app/src/main/java/org/schabi/newpipe/streams/io/SharpInputStream.java

```diff
@@ -0,0 +1,48 @@
+package org.schabi.newpipe.streams.io;
+
+import androidx.annotation.NonNull;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+public class SharpInputStream extends InputStream {
```

Comment:
Could you add an explanation what the difference of this and `SharpOutputStream` to normal java streams is and why it is needed. The `SharpStream` class only tells me that it's from C#

Reply:
Ok, I will try to investigate it myself, since I don't know ;-)

Reply:
SharpStream is a wrapper around the 2 different APIs for SAF and non-SAF. It has both input and output like in C#, while in Java those are usually different classes. Because a class could only extend one class, there is SharpInputStream and SharpOutputStream that wraps SharpStream while extending InputStream or OutputStream, so that you could use it with built-in Java stuff that supports InputStream or OutputStream.